src/components/PopupDetails.tsx

```diff
@@ -52,9 +57,180 @@ const PopupDetails: React.FC<PopupDetailsProps> = ({
 
   // const { settings, updateSettings } = useSettings();
   const [isEnabled, setIsEnabled] = useState(point.isEnabled);
+  type EditableField = "id" | "ssid" | "bssid";
+
+  const [editableValues, setEditableValues] = useState<
+    Record<EditableField, string>
+  >({
+    id: point.id,
+    ssid: point.wifiData?.ssid ?? "",
+    bssid: formatMacAddress(point.wifiData?.bssid ?? ""),
+  });
+  const [currentPointId, setCurrentPointId] = useState(point.id);
+  const [hasIdError, setHasIdError] = useState(false);
+  const idInputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    setIsEnabled(point.isEnabled);
+    setCurrentPointId(point.id);
+    setHasIdError(false);
+    setEditableValues({
+      id: point.id,
+      ssid: point.wifiData?.ssid ?? "",
+      bssid: formatMacAddress(point.wifiData?.bssid ?? ""),
+    });
+  }, [point]);
+
+  const handleInputChange = (field: EditableField) => {
+    return (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+      setEditableValues((prev) => ({ ...prev, [field]: value }));
+      if (field === "id") {
+        setHasIdError(false);
+      }
+    };
+  };
+
+  const handleInputBlur = (field: EditableField) => {
+    return (_event: React.FocusEvent<HTMLInputElement>) => {
+      if (field === "id") {
+        const newId = editableValues.id.trim();
+        if (newId.length === 0) {
+          setEditableValues((prev) => ({ ...prev, id: currentPointId }));
+          setHasIdError(true);
+          setTimeout(() => {
+            idInputRef.current?.focus();
+            idInputRef.current?.select();
```

Comment:
I would be pretty annoyed by this btw. I know it's subjective, but I think it's not a good UX. Would just show the error and not focus, but I leave this decision to you